test/compiler.test.js

```diff
@@ -1195,6 +1339,63 @@ describe('compiler', function() {
         }
       });
     });
+
+    it('loads component relative to appRootDir', function() {
+      appdir.writeConfigFileSync('./component-config.json', {
+        './component': { }
+      });
+      var appJS = appdir.writeConfigFileSync('component.js', '');
+
+      var instructions = boot.compile(appdir.PATH);
+      expect(instructions.components[0]).have.property(
+        'sourceFile', appJS
+      );
+    });
+
+    it('loads component relative to node modules', function() {
+      appdir.writeConfigFileSync('component-config.json', {
+        'index': { }
+      });
+      var appJS = appdir.writeConfigFileSync('node_modules/index.js', '');
```

Comment:
It is rather unusual to have `node_modules/index.js`, typically the script is nested in a module directory, e.g. `node_modules/mycomponent/index.js`. Please rework the test to use `mycomponent` as the key in the config file and check that is is resolved to `node_modules/mycomponent/index.js`.

Please make the same change in the coffescript test below too.